File: bwl.Imaging/Tools/Extensions.vb
Imports System.Runtime.CompilerServices

Public Module Extensions
    <Extension()>
    Public Function ToDrawingPoint(point As PointF) As System.Drawing.Point
        Return New System.Drawing.Point(point.X, point.Y)
    End Function

    <Extension()>
    Public Function Dist(this As PointF, arg As PointF) As Single
        Return Math.Sqrt((this.X - arg.X) ^ 2 + (this.Y - arg.Y) ^ 2)
    End Function

    <Extension()>
    Public Function ToString(this As PointF) As String
        Return "X: " + this.X.ToString() + " Y: " + this.Y.ToString
    End Function
End Module

File: Bwl.Imaging.Skia/MatrixProcessing/Extensions.cs
using SkiaSharp;

namespace Bwl.Imaging.Skia;

public static class Extensions
{
    public static SKPointI ToPoint(this SKPoint pointF)
    {
        return new SKPointI((int)Math.Round(pointF.X), (int)Math.Round(pointF.Y));
    }

    public static SKPoint ToPointF(this SKPointI point)
    {
        return new SKPoint(point.X, point.Y);
    }

    public static float Dist(this SKPoint @this, SKPoint arg)
    {
        return (float)Math.Sqrt(Math.Pow((double)(@this.X - arg.X), 2d) + Math.Pow((double)(@this.Y - arg.Y), 2d));
    }

    public static string ToString(this SKPoint @this)
    {
        return "X: " + @this.X.ToString() + " Y: " + @this.Y.ToString();
    }

    public static SKRect ToPositiveSized(this SKRect @this)
    {
        float left = Math.Min(@this.Left, @this.Right);
        float right = Math.Max(@this.Left, @this.Right);
        float top = Math.Min(@this.Top, @this.Bottom);
        float bottom = Math.Max(@this.Top, @this.Bottom);
        return new SKRect(left, top, right, bottom);
    }

    public static SKRectI ToPositiveSized(this SKRectI @this)
    {
        int left = Math.Min(@this.Left, @this.Right);
        int right = Math.Max(@this.Left, @this.Right);
        int top = Math.Min(@this.Top, @this.Bottom);
        int bottom = Math.Max(@this.Top, @this.Bottom);
        return new SKRectI(left, top, right, bottom);
    }
}